Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIExtensionCocoa.mm

```diff
@@ -46,12 +51,36 @@
 
 NSURL *WebExtensionAPIExtension::getURL(NSString *resourcePath, NSString **errorString)
 {
-    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extension/getURL
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/extension/getURL
 
     URL baseURL = extensionContext().baseURL();
     return resourcePath.length ? URL { baseURL, resourcePath } : baseURL;
 }
 
+bool WebExtensionAPIExtension::isInIncognitoContext(WebPage* page)
+{
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/extension/inIncognitoContext
+
+    return page->usesEphemeralSession();
+}
+
+void WebExtensionAPIExtension::isAllowedFileSchemeAccess(Ref<WebExtensionCallbackHandler>&& callback)
+{
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/extension/isAllowedFileSchemeAccess
+    // FIXME: rdar://problem/58428135 Consider allowing file URL access if the user opted in explicitly in some way.
```

Comment:
Should this be a Bugzilla bug instead?

Reply:
I kept it as a radar since we already have it an it is more of a policy decision that will need addressed before it can be fixed, which is more internal centric.